uisdk/src/main/java/tech/dojo/pay/uisdk/domain/mapper/PaymentIntentDomainEntityMapper.kt

```diff
@@ -38,6 +39,9 @@ internal class PaymentIntentDomainEntityMapper {
             orderId = raw.reference ?: "",
             isSetUpIntentPayment = !raw.merchantInitiatedType.isNullOrBlank() && !raw.paymentSource.isNullOrBlank(),
             merchantName = raw.config?.tradingName ?: "",
+            isPaymentAlreadyCollected =
+            PaymentIntentStatusDomainEntity.fromStatus(requireNotNull(raw.status)) == PaymentIntentStatusDomainEntity.CAPTURED ||
+                PaymentIntentStatusDomainEntity.fromStatus(requireNotNull(raw.status)) == PaymentIntentStatusDomainEntity.AUTHORIZED,
```

Comment:
these `requireNotNull` will throw an exception, is that correct?

Reply:
yes before we rich that point we call this `checkInvalidParameters` which will check all the needed parameter and one of them is  `status ` and if it fails it will `throw EssentialParamMissingException(invalidParams)` which will be handled

Reply:
yeah, but I don't think that's a good approach thinking in a long term. Whoever that use this method will have to know somehow that is possible to throw an example from here. I would rather return null values whenever those values are null rather than throwing an exception.

Reply:
this method is already used on `handleSuccessPaymentIntent` which got a try catch  and in that case it maps to `FetchFailure` so it's already handled

Reply:
Talked with @Saamyyy , to be fixed in a next PR